File: cache/src/main/java/com/mm/http/HOST.java
package com.mm.http;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * Created by : majian
 * Date : 2021/8/24
 */

@Retention(RetentionPolicy.RUNTIME)
@Target(ElementType.TYPE)
public @interface HOST {
    String value(); // host for service
}

File: cache/src/main/java/com/mm/http/HOST.kt
package com.mm.http

/**
 * Created by : majian
 * Date : 2021/8/24
 * 业务需求需要,默认 hostType 为 DOMAIN_URL。
 * @param value host for service
 * @param hostType  host type for service
 */
@Retention(AnnotationRetention.RUNTIME)
@Target(AnnotationTarget.ANNOTATION_CLASS, AnnotationTarget.CLASS)
annotation class HOST(val value: String, val hostType: Int = HOST.NORMAL) {

    companion object {
        const val NORMAL = 0 // URL 默认的使用value值作为host，其他类型自行添加
    }
}